vertical-pod-autoscaler/pkg/admission-controller/logic/limitrange_checker.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logic
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+
+	v1_listers "k8s.io/client-go/listers/core/v1"
+)
+
+// LimitsHints provide hinted limits that respect limit range max ratio
+type LimitsHints interface {
+	IsNil() bool
+	RequestsExceedsRatio(indexOfContainer int, resourceName v1.ResourceName) bool
+	HintedLimit(indexOfContainer int, resourceName v1.ResourceName) resource.Quantity
+}
+
+// LimitRangeHints implements LimitsHints interface
+type LimitRangeHints struct {
+	requestsExceedsRatio  []map[v1.ResourceName]bool
```

Comment:
since v1.ResourceList is already a map, can we store proposedLimits map[container_name]proposed_limit_changes
Then:

requestExceedsRatio(container_name, resourceName) bool {
  _, found := proposedLimits[container_name][resourceName]
  return found
}